src/logger.cpp

```diff
@@ -85,7 +85,11 @@ size_t Logger::errorCount() const
 
 ErrorPtr Logger::getError(size_t index) const
 {
-    return mPimpl->mErrors.at(index);
+    ErrorPtr err = nullptr;
+    if (index < mPimpl->mErrors.size()) {
+        err = mPimpl->mErrors.at(index);
+    }
+    return err;
```

Comment:
I like that form but it does hide branches from the coverage testing.  As I (hope I) have written all functions to follow this pattern I think we should leave it as is.